src/qinfer/distributions.py

```diff
@@ -321,6 +322,76 @@ def sample(self, n=1):
         ), len(cumsum_weights) - 1)]
 
     ## MOMENT FUNCTIONS ##
+    @staticmethod
+    def particle_mean(weights, locations):
+        r"""
+        Returns the mean of a function :math:`f` over model
```

Comment:
I think this particular function does not support expectations over a function f? Should we somehow tie it in with est_mean_fn?